pysmt/constants.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+
+# Copyright 2014 Andrea Micheli and Marco Gario
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from six import PY2
+# The environment variable can be used to force the configuration
+# of the Fraction class.
+#
+# If it is not set, the default behavior is to use GMPY2 if they are
+# installed.
+#
+# If set to True, GMPY2 are required and and exception will be thrown
+# if they are not available
+#
+# If set to True, Python's Fraction module will be used instead.
+#
+
+ENV_USE_GMPY = os.environ.get("PYSMT_GMPY")
+if ENV_USE_GMPY is not None:
+    ENV_USE_GMPY = ENV_USE_GMPY.lower() in ["true", "1"]
+
+HAS_GMPY = False
+try:
```

Comment:
No (see comment below): we import them to check the mpq/mpz type. You are right below, these types should probably be defined here.